File: Leave-Management(Angular-Spring)/Frontend/src/app/models/employee.ts
export class employee {
    email: string = ''
    password: string = ''
    name: string = ''
    lastname: string = ''
    gender: 'Male' | 'Female' | null = null;
    role: 'admin' | 'employee' = 'employee';
    phone:string=''
    job: string = ''
    leave_type:string=''
    leave_start!:Date
    leave_end!:Date
    leaveDecision: 'approved' | 'declined' | null = null;
}

File: Backend-spring-angular/src/main/java/com/example/Backend/spring_angular/model/Employee.java
package com.example.Backend.spring_angular.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "employees")
@Getter  // Lombok generates all getters
@Setter  // Lombok generates all setters
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String lastname;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String gender;

    @Column(nullable = false)
    private String role = "employee";

    @Column(nullable = false)
    private String job;

    @Column
    private String leave_type;

    @Column
    private Date leave_start;

    @Column
    private Date leave_end;

    @Column
    private String leave_decision;
}
